pkg/channels/channels.go

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package channels
+
+import "sync"
+
+type DoneChan <-chan struct{}
+
+var ClosedDoneChan DoneChan
+
+func init() {
+	ch := make(chan struct{})
+	close(ch)
+	ClosedDoneChan = ch
+}
+
+// Merge merges a slice of channels and returns a channel that is
+// closed when all the merged channels are closed.
+func Merge[T any](cs ...<-chan T) <-chan T {
+	out := make(chan T)
+	var wg sync.WaitGroup
+	wg.Add(len(cs))
+	for _, c := range cs {
+		c := c
```

Comment:
nit: with Go 1.22 this is no longer needed, see https://go.dev/doc/go1.22#language and https://go.dev/wiki/LoopvarExperiment.

Reply:
Good point, thanks! :+1: